x-pack/plugin/security/qa/jwt-realm/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRestIT.java

```diff
@@ -499,6 +508,57 @@ public void testAuthenticationFailureIfDelegatedAuthorizationFails() throws Exce
         }
     }
 
+    public void testReloadClientSecret() throws Exception {
```

Comment:
I'm using this new test as a test of the underlying infrastructure; the exact details of JWT authentication are not very important here. The gist is that there is a reloadable secure setting that is compared against a header passed in as part of JWT request authc. This test primarily demonstrates that setting it to a new value and reloading (without a cluster restart) works as expected.